frontend/src/pages/PipelineDetails.test.tsx

```diff
@@ -89,7 +93,12 @@ describe('PipelineDetails', () => {
     deletePipelineSpy.mockReset();
     getPipelineSpy.mockImplementation(() => Promise.resolve(testPipeline));
     getPipelineSpy.mockClear();
-    getTemplateSpy.mockImplementation(() => Promise.resolve('{}'));
+    getRunSpy.mockImplementation(() => Promise.resolve(testRun));
```

Comment:
Are you just using `Promise.resolve` to make it closer to the actual implementation? (i.e. returning a promise?)

Reply:
There's no real need for it here. The only difference I can think of is that this catches the case where the calling code mistakenly doesn't use `await` when calling this function. I realize this isn't the case in all tests though, but I think this is more correct. wdyt?

Reply:
Is await actually needed for the spies though? If it is, then I think your solution is an adequate one

Reply:
The component's code should be calling `await` on the API. If it's not, it's a bug, and this catches that.